File: android/src/main/java/org/axen/flutter/glide/provider/GifImageProvider.java
package org.axen.flutter.glide.provider;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestBuilder;
import com.bumptech.glide.load.resource.gif.GifDrawable;

public final class GifImageProvider extends AbstractProvider<GifDrawable> {

    public GifImageProvider(@NonNull Context context) {
        super(context);
    }

    @NonNull
    protected RequestBuilder<GifDrawable> build() {
        return Glide.with(this.context).asGif();
    }

}

File: android/src/main/kotlin/org/axen/flutter/glide/provider/GifImageProvider.kt
package org.axen.flutter.glide.provider

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestBuilder
import com.bumptech.glide.load.resource.gif.GifDrawable
import org.axen.flutter.glide.common.AbstractProvider

class GifImageProvider(context: Context): AbstractProvider<GifDrawable>(context) {

    override fun build(): RequestBuilder<GifDrawable> {
        return Glide.with(context).asGif()
    }
}